packages/api/endpoints/collections.js

```diff
@@ -197,32 +198,42 @@ async function put(req, res) {
   }
 
   try {
-    oldCollection = await collectionsModel.get({ name, version });
+    oldPgCollection = await collectionPgModel.get(knex, { name, version });
   } catch (error) {
     if (error.name !== 'RecordDoesNotExist') {
       throw error;
     }
     return res.boom.notFound(`Collection '${name}' version '${version}' not found`);
   }
 
+  try {
+    oldCollection = await collectionsModel.get({ name, version });
+  } catch (error) {
+    if (error.name !== 'RecordDoesNotExist') {
+      throw error;
+    }
+    log.warn(`Dynamo record for Collection '${name}' version '${version}' not found, proceeding to update with postgresql record alone`);
+  }
+
   collection.updatedAt = Date.now();
-  collection.createdAt = oldCollection.createdAt;
+  collection.createdAt = oldPgCollection.created_at.getTime();
 
   const postgresCollection = translateApiCollectionToPostgresCollection(collection);
 
   try {
     await createRejectableTransaction(knex, async (trx) => {
-      const [pgCollection] = await collectionPgModel.upsert(trx, postgresCollection);
-      const translatedCollection = await translatePostgresCollectionToApiCollection(pgCollection);
+      await collectionPgModel.upsert(trx, postgresCollection);
       dynamoRecord = await collectionsModel.create(collection);
       // process.env.ES_INDEX is only used to isolate the index for
       // each unit test suite
       await indexCollection(esClient, dynamoRecord, process.env.ES_INDEX);
-      await publishCollectionUpdateSnsMessage(translatedCollection);
+      await publishCollectionUpdateSnsMessage(collection);
```

Comment:
All the same, it might be better to publish `dynamoRecord` that comes back from `create()` than the `collection` which was sent for writing

Reply:
Hmm, solid comment. 

I....agree with the concern that conditional write logic in the model should be accommodated, but as we move away from Dynamo, that model logic shouldn't be authoritative/will be removed in favor of whatever the db write logic is.  

 I'm wondering if the correct pattern here more generally is to address this concern by using the returned *postgres* record to publishCollectionUpdateSnsMessage, as that pattern works as the translation method doesn't *currently* invoke a database call.    In other endpoints that would be problematic.  🤔 
 
 *frown* 

There's potential for adopting a different pattern/refactor here in phase 3 we should consider in making sure that db model logic makes it back to the publish call *without* incurring further load on the db when possible, but I think your suggestion isn't problematic/is more correct pending phase 3 work.  

00a2b119b

Reply:
> hat model logic shouldn't be authoritative/will be removed in favor of whatever the db write logic is.

I'm not sure what this means. There still could be conditional logic in the PG writes , which means translating before publishing to SNS is the only option. But I agree that this is fine for now.

Reply:
Right, sorry that wasn't clear.   Right now the sns post is ignoring anything in the db model as it's just using the *dynamo* model.   That's not what we want in Phase 3, but the work to deal with the api->postgres->api format without inefficient db loading should be done there, and this is ok for now.  👍🏻